src/components/TodoApp.tsx

```diff
@@ -0,0 +1,133 @@
+import { useContext, useEffect, useState } from 'react';
+
+import { TodosContext } from '../TodosContext';
+import TodoList from './TodoList';
+import TodosFilter from './TodosFilter';
+import { Todo } from '../types/Todo';
+import { Status } from '../enums/Status';
+
+const filteredTodos = (todos: Todo[], filter: Status): Todo[] => {
+  switch (filter) {
+    case Status.Active:
+      return todos.filter(todo => !todo.completed);
+    case Status.Completed:
+      return todos.filter(todo => todo.completed);
+    default:
+      return todos;
+  }
+};
+
+const TodoApp: React.FC = () => {
+  const { state, dispatch } = useContext(TodosContext);
```

Comment:
- You can create a hook, so you don't need to pass a context every time. `const useTodos = () => useContext(TodosContext);`
- Also, `state/dispatch` are not informative, specify names as you can have more than one reducer